.github/workflows/lint.yml

```
@@ -0,0 +1,30 @@
+name: Lint (flake8)
+
+# Controls when the workflow will run
+on: [push]
+
+permissions:
+  contents: read
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v3
+        with:
+          python-version: "3.8"
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install flake8 pytest
+          if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+      - name: Lint with flake8
+        run: |
+          # stop the build if there are Python syntax errors or undefined names
+          flake8 . --count --select=E9,F63,F7,F82,F401 --show-source --statistics
+          # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+          flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
```

@elraphty , I suggest we move those into a `.flake8` config file to the repository, like this
```ini
[flake8]
max-line-length = 127
max-complexity = 10
select = E9,F63,F7,F82,F401
statistics = True
count = True
show-source = True
```

We could then use this same configuration file in our GitHub Actions workflow for consistency.
```suggestion
         flake8 .
```

This approach ensures consistency between local checks and CI, makes it easier for contributors to verify their code before pushing, and centralizes our Flake8 configuration. What do you think?

---

> @elraphty , I suggest we move those into a `.flake8` config file to the repository, like this
> 
> ```ini
> [flake8]
> max-line-length = 127
> max-complexity = 10
> select = E9,F63,F7,F82,F401
> statistics = True
> count = True
> show-source = True
> ```
> 
> We could then use this same configuration file in our GitHub Actions workflow for consistency.
> 
> This approach ensures consistency between local checks and CI, makes it easier for contributors to verify their code before pushing, and centralizes our Flake8 configuration. What do you think?

yeah makes sense, flake is for local check before making a PR, I will add this thanks

---

@kouloumos, I have added Flake8 and Pre commit